lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java

```diff
@@ -1158,6 +1158,10 @@ private Task claim(String taskId, boolean forceClaim)
     try {
       taskanaEngine.openConnection();
       task = (TaskImpl) getTask(taskId);
+
+      TaskImpl oldTask = task.copy();
+      oldTask.setId(taskId);
+      oldTask.setExternalId(task.getExternalId());
```

Comment:
I am not a fan of creating the `oldTask` this way. How do we do it for different operations?

Reply:
createTask and updateTask are methods which also use the determineChangesInAttributes-method. However, these two get a Task object as a parameter and then perform a getTask to save the old state of the task before changing.
See:
```

 public Task updateTask(Task task)
      throws InvalidArgumentException, TaskNotFoundException, ConcurrencyException,
          NotAuthorizedException, AttachmentPersistenceException,
          ObjectReferencePersistenceException, InvalidStateException,
          ClassificationNotFoundException {
    String userId = taskanaEngine.getEngine().getCurrentUserContext().getUserid();
    TaskImpl newTaskImpl = (TaskImpl) task;
    try {
      taskanaEngine.openConnection();
      TaskImpl oldTaskImpl = (TaskImpl) getTask(newTaskImpl.getId());
```

claim and cancelClaim have a taskId as parameter. I tried to call getTask two times, so I can perform changes on one object and save the state of the object with the other one. The problem is that calling getTask behind each toher on the same id will return the same object (same object reference). So no changes are detected with the determineChangesInAttributes-method. I confirmed this behaviour together with @gitgoodjhe . MyBatis could be the reason for this.
To conclude, I couldn't think of a better/easier way of storing the old state of the object.

Reply:
Alright. Thanks for the clarification.